From 1de47b8cbea8e8e486941830d5963c7e52750b67 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Tue, 11 Jul 2023 22:19:07 +0100
Subject: [PATCH 1/1] ypkg2: Only enable fakeroot during the install & check
 steps

This will only allow $installdir to be used during install and check steps.
(Though, packagers should only be using it during install.)

This is extremely desirable due to the extreme overhead of fakeroot.
Especially when ccache is enabled but there are no cache hits for a build.

Unsure how yet if this will break builds in subtle ways. If needed, we can
add a context key to enable/disable this behavior.
---
 ypkg2/main.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/ypkg2/main.py b/ypkg2/main.py
index 5d5ccde..4f80394 100644
--- a/ypkg2/main.py
+++ b/ypkg2/main.py
@@ -148,6 +148,12 @@ def execute_step(context, step, step_n, work_dir):
     if context.avx2 and step_n == "install":
         endScript = "%avx2_lib_shift"
 
+    # We only want fakeroot for the install and check steps due
+    # to massive performance overhead.
+    # This is a quick n' dirty way of achieving that
+    if not step_n in ["install", "check"]:
+        script.define_unexport("LD_PRELOAD")
+
     exports = script.emit_exports()
 
     # Run via bash with enable and error
-- 
2.40.1

